src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/parser/SqlParserTest.java

```diff
@@ -1116,6 +1116,36 @@ public void caseWhenTestWithouhtElseExpr() throws SqlParseException {
 
     }
 
+    @Test
+    public void caseWhenSwitchTest() throws SqlParseException {
+        String query = "SELECT CASE weather "
+                + "WHEN 'Sunny' THEN '0' "
+                + "WHEN 'Rainy' THEN '1' "
+                + "ELSE 'NA' END AS case "
+                + "FROM t";
+        SQLExpr sqlExpr = queryToExpr(query);
+        Select select = parser.parseSelect((SQLQueryExpr) sqlExpr);
+        for (Field field : select.getFields()) {
```

Comment:
I'm really suspicuous about this test. Can we get rid of logic and evaluate assertions as much as we can?

Reply:
Yes, both here and the atomic integer is what i referred from the previous code. I was also suspicious about the necessity to do it in parallel since we don't touch any concurrent computing in our plugin. I will modify the test.